backend/user/views.py

```diff
@@ -109,14 +119,32 @@ def post(self, request):
                 notificationsetting__enabled=True,
             )
             .exclude(token="")
+            .values_list("user__username", "token", "dev")
         )
 
+        # unpack list of tuples
+        if tokens:
+            success_users, tokens, devs = zip(*tokens)
+        else:
+            success_users, devs = list(), list()
+        failed_users = list(set(usernames) - set(success_users))
+
         if len(tokens) == 1:
-            send_push_notif(tokens[0], title, body, isDev)
+            if delay:
+                send_delayed_push_notif(tokens[0], title, body, delay, devs[0])
+            else:
+                send_push_notif(tokens[0], title, body, devs[0])
         elif len(tokens) > 1:
-            send_push_notif_batch(tokens, title, body, isDev)
+            # ensure that tokens are the same type
+            if any(dev != devs[0] for dev in devs):
```

Comment:
I think this line might be a bit slow (10000+ tokens). Also, this line will fail for any batch notification that includes Jongmin, because he has 2 notification tokens (one dev and one not) and not sure if your query differentiates between that. I would suggest removing this line and making the query on L114 filter for a `dev=False`, unless otherwise specified.